### ISSUE TITLE
Deterministically generate transaction key

### ISSUE THREAD

**Issue by philogy** [monero-project/monero#6638]
After transactions are created it may still be useful to know the transaction key to prove payment to an auditor for example. A problem with Monero wallets at the moment is that transaction keys are created completely randomly meaning upon loss of the wallet and restoration from a mnemonic seed the transaction keys are permanently lost. This could be remedied by generating the transaction key deterministically from wallet parameters and input parameters leading to a fully recoverable wallet.

**Comment by philogy** [comment]
Well yes you can use the private transaction key to prove you created a payment without having to reveal your private view key for example

**Comment by SarangNoether** [comment]
If properly implemented, this would also have the benefit of mitigating against poor random number generation. A bad random number generator would negatively affect other areas of transaction construction, of course.

**Comment by UkoeHB** [comment]
My concern is if a person's private keys are compromised and tx private keys are deterministic, then the compromiser will be able to figure out the recipients of all his transactions. 

Moreover, knowing tx private keys isn't enough to prove payment to an auditor, as the recipient addresses are also required. Since recipient addresses must be stored anyway, not much is gained/lost whether or not the tx private keys are stored.

**Comment by moneromooo-monero** [comment]
You can prove payment without having to give a tx key.

**Comment by SarangNoether** [comment]
Right, but a proof of knowledge of the transaction key is one simple way that also has the benefit of reducing reliance on random number generation.

**Comment by philogy** [comment]
Don't you have to reveal your private view key to be able to prove payment with the tx key?

**Comment by SarangNoether** [comment]
There are different types of transaction proofs available. You could sign messages with the transaction private key that can be verified with only the transaction public key, without revealing secret data.

**Comment by philogy** [comment]
Yeah I mistyped I meant don't you have to reveal your private view key to be able to prove payment **without** the transaction private key?

**Comment by SarangNoether** [comment]
There are other things you can prove without it, like being able to sign using the output private key against the key image (which reveals the public key), or generating a new signature using the same ring and key image.

**Comment by stoffu** [comment]
> or generating a new signature using the same ring and key image.

FWIW, this is what spend proofs do (see `get/check_spend_proof` command).

**Comment by UkoeHB** [comment]
Ch. 8 of [ZtM2](https://web.getmonero.org/library/Zero-to-Monero-2-0-0.pdf) discusses all the transaction proofs in Monero (and some that don't exist yet).
